tests/nat/utils.ts

```diff
@@ -0,0 +1,1164 @@
+import type { ChildProcess } from 'child_process';
+import os from 'os';
+import fs from 'fs';
+import path from 'path';
+import process from 'process';
+import child_process from 'child_process';
+import readline from 'readline';
+import Logger, { LogLevel, StreamHandler } from '@matrixai/logger';
+import * as testBinUtils from '../bin/utils';
+
+type NATType = 'eim' | 'edm' | 'dmz' | 'edmSimple';
+
+// Constants for all util functions
+// Veth pairs (ends)
+const agent1Host = 'agent1';
+const agent2Host = 'agent2';
+const agent1RouterHostInt = 'router1-int';
+const agent1RouterHostExt = 'router1-ext';
+const agent2RouterHostInt = 'router2-int';
+const agent2RouterHostExt = 'router2-ext';
+const router1SeedHost = 'router1-seed';
+const router2SeedHost = 'router2-seed';
+const seedRouter1Host = 'seed-router1';
+const seedRouter2Host = 'seed-router2';
+// Subnets
+const agent1HostIp = '10.0.0.2';
+const agent2HostIp = '10.0.0.2';
+const agent1RouterHostIntIp = '10.0.0.1';
+const agent2RouterHostIntIp = '10.0.0.1';
+const agent1RouterHostExtIp = '192.168.0.1';
+const agent2RouterHostExtIp = '192.168.0.2';
+const router1SeedHostIp = '192.168.0.1';
+const seedHostIp = '192.168.0.3';
+const router2SeedHostIp = '192.168.0.2';
+// Subnet mask
+const subnetMask = '/24';
+// Ports
+const agent1Port = '55551';
+const agent2Port = '55552';
+const mappedPort = '55555';
```

Comment:
These should be notated as constants. We usually our constants like this:

```
const ABC_DEF = 'blah';
```

Also use docblock comments for these. You can use something like this:

```
/**
 * ...
 */
```